cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -614,6 +615,11 @@ public boolean hasType(String datasetTypeName) {
   }
 
   private void publishAudit(Id.DatasetInstance datasetInstance, AuditType auditType) {
+    // Don't publish audit for system datasets admin operations, there can be a deadlock
```

Comment:
Tracker is only one system that is used to viusalize audit messages right? Audit messages are for the whole CDAP system. I think it is okay to publish audit messages for system entities as long as it does not cause any issue. But since we don't publish audit in standalone mode now, we can revisit this. JIRA - https://issues.cask.co/browse/CDAP-5447